### PR TITLE
Generate inverse relation names

### LINKED ISSUE #54: Model generator, add inverseOf to relations.
I've just implemented this in my model generator. Filling those manually is very error prone (singular vs plural relation names, etc). This be included in core but I've noticed that not all relations need those (excluding many-many). I'd rather remove those unecessary than try to add the required ones.

Should there be an option to generate them, like _generateRelations_? Maybe _generateRelations_ should be a dropdown with options like:

* no relations
* generate relations
* generate relations with inverse

### PR DESCRIPTION
**Pull request by nineinchnick**
Resolves #54.

Expands the boolean _generateRelations_ option into a dropdown, adding a new option to generate inverse relation names.

### CODE REVIEW

File: generators/model/Generator.php [hunk 1]
```diff
@@ -463,10 +466,76 @@ protected function generateRelations()
             }
         }
 
+        if ($this->generateRelations === self::RELATIONS_ALL_INVERSE) {
+            return $this->generateInverseRelations($relations);
```
Comment (samdark) [inline generators/model/Generator.php]: Is it better to name it `addInverseRelations`?

File: generators/model/Generator.php [hunk 2]
```diff
@@ -463,10 +466,76 @@ protected function generateRelations()
...
+    }
+
+    /**
+     * @param array $relations relation declarations
+     * @return array relation declarations extended with inverse relation names
+     */
+    protected function generateInverseRelations($relations)
+    {
+        $relationNames = [];
+        $db = $this->getDbConnection();
```
Comment (samdark) [inline generators/model/Generator.php]: It's used only once. No need for a variable.

File: generators/model/Generator.php [hunk 3]
```diff
@@ -463,10 +466,76 @@ protected function generateRelations()
...
+        }
         return $relations;
     }
 
     /**
+     * @param TableSchema $table
+     * @param array $fks
+     * @return bool
+     */
+    protected function isManyRelation($table, $fks)
```
Comment (samdark) [inline generators/model/Generator.php]: Is it better to name it `isHasManyRelation`?

Comment (nineinchnick) [inline generators/model/Generator.php]: I'm not sure. Having two verbs at the beginning is a bit confusing.

Comment (samdark) [inline generators/model/Generator.php]: Well, relation type is `hasMany`, not `many`...

Comment (nineinchnick) [inline generators/model/Generator.php]: It's stupid to argue about it, but I only found mentions of  [`one-to-many`](https://en.wikipedia.org/wiki/One-to-many_%28data_model%29) on Wikipedia :smile: 

Anyway, for consistency and whatever else needed to get this merged, I renamed it.

Comment (samdark) [inline generators/model/Generator.php]: Yes. Classic ER types are `one-to-many`, `one-to-one`, `many-to-many` but that's not the case for Yii 2.0 ;)

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii2-gii/commit/a67035cd3225ea108db03a4da21b0204fe54630d. Thanks!

**Pull request closed without merging**